### PR TITLE
Set Config via QR Code

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/10311/set-partner-config-via-qr-code-on-sample-app

## Summary

Allow partners to scan QR code from portal and fill in the smile config automatically

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt [hunk 1]
```diff
@@ -37,6 +39,8 @@ fun SmileConfigModalBottomSheet(
     dismissable: Boolean = true,
 ) {
     var configInput by remember { mutableStateOf("") }
+    val context = LocalContext.current
+    val scanner = GmsBarcodeScanning.getClient(context)
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt:43]: Does this need to be wrapped in a `remember`? Something like
```suggestion
    val scanner by remember(context) { GmsBarcodeScanning.getClient(context) }
```

File: sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt [hunk 2]
```diff
@@ -52,6 +56,18 @@ fun SmileConfigModalBottomSheet(
                 .padding(16.dp),
             horizontalAlignment = Alignment.CenterHorizontally,
         ) {
+            Button(
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt:59]: Can we make this in-line with the save config button below? And make it an OutlinedButton to delineate it from the Save button

File: sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt [hunk 3]
```diff
@@ -52,6 +56,18 @@ fun SmileConfigModalBottomSheet(
...
+                modifier = Modifier.fillMaxWidth(),
+                onClick = {
+                    scanner.startScan()
+                        .addOnSuccessListener { barcode ->
+                            barcode.rawValue?.let { onSaveSmileConfig(it) }
+                        }
+                },
+            ) {
+                Text(text = stringResource(id = R.string.scan_qr_code))
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt:68]: not necessary, but would be a nice touch to include a small QR code logo as well

File: sample/sample.gradle.kts [hunk 4]
```diff
@@ -146,6 +146,9 @@ dependencies {
 
     implementation(libs.androidx.navigation.compose)
 
+    // QR Scanner
```
Comment (vanshg) [inline sample/sample.gradle.kts:149]: nit: i think we could do without this comment
```suggestion
```

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -37,6 +37,7 @@ play-services-mlkit-face-detection = "17.1.0"
 retrofit = "2.9.0"
 sentry = "6.28.0"
 timber = "5.0.1"
+qr-code-scanner = "16.1.0"
```
Comment (vanshg) [inline gradle/libs.versions.toml:40]: 2 things:
- can we rename this to `mlkit-code-scanner`? (this is to keep the same convention as the mlkit dependency that was added [here](https://github.com/smileidentity/android/pull/208/files#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR31))
- Can you move this to be at the right place in this alphabetical list?

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -99,5 +100,6 @@ retrofit-converter-moshi = { module = "com.squareup.retrofit2:converter-moshi",
 sentry = { module = "io.sentry:sentry" }
 sentry-bom = { module = "io.sentry:sentry-bom", version.ref = "sentry" }
 timber = { module = "com.jakewharton.timber:timber", version.ref = "timber" }
+qr-code-scanner = { module = "com.google.android.gms:play-services-code-scanner", version.ref = "qr-code-scanner"}
```
Comment (vanshg) [inline gradle/libs.versions.toml:103]: ditto

File: sample/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -14,6 +14,12 @@
             android:enableOnBackInvokedCallback="true"
             android:theme="@style/Theme.SmileID"
             tools:targetApi="33">
+
+        <meta-data
+                android:name="com.google.mlkit.vision.DEPENDENCIES"
+                android:value="face,barcode_ui"
+                tools:replace="android:value"/>
```
Comment (vanshg) [inline sample/src/main/AndroidManifest.xml:21]: Can you add a comment about why we needed to add `tools:replace`?

**Review by vanshg — CHANGES_REQUESTED** [review]

File: sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt [hunk 8]
```diff
@@ -37,6 +43,8 @@ fun SmileConfigModalBottomSheet(
     dismissable: Boolean = true,
 ) {
     var configInput by remember { mutableStateOf("") }
+    val context = LocalContext.current
+    val scanner = remember(context) { GmsBarcodeScanning.getClient(context) }
```
[the remark this answers is not in the document]
Comment (jumaallan) [reply sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt:47]: I don't think so, unless I missed something

The builder is where we set the options

[the remark this answers is not in the document]
Comment (vanshg) [reply sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt:47]: Looking at step 2:

```kotlin
val scanner = GmsBarcodeScanning.getClient(this)
// Or with a configured options
// val scanner = GmsBarcodeScanning.getClient(this, options)
```

File: sample/src/main/java/com/smileidentity/sample/SmileIDApplication.kt [hunk 9]
```diff
@@ -15,6 +17,19 @@ class SmileIDApplication : Application() {
         // *****Note to Partners*****
         // The line below is how you should initialize the SmileID SDK
         // SmileID.initialize(this)
+
+        // Initialize bar code scanner
+        initializeBarcodeScanner()
+    }
+
+    private fun initializeBarcodeScanner() {
+        GmsBarcodeScannerOptions.Builder()
```
[the remark this answers is not in the document]
Comment (vanshg) [reply sample/src/main/java/com/smileidentity/sample/SmileIDApplication.kt:26]: This doesn't need to be done at the application's onCreate. It needs to be done when creating the client in the bottom sheet

File: gradle/libs.versions.toml [hunk 10]
```diff
@@ -27,6 +27,7 @@ kotlin-immutable-collections = "0.3.5"
 ksp = "1.9.10-1.0.13"
 ktlint-plugin = "11.5.1"
 leakcanary = "2.12"
+mlkit-code-scanner = "16.1.0"
```
Comment (vanshg) [inline gradle/libs.versions.toml:30]: This should go after `maven-publish`

**Comment by vanshg** [comment]
I just tried this in landscape mode, and the QR code button gets cut off. 2 things we can do to fix -- reduce inter-component spacing to 4 or 8 dp. And add vertical scroll to the column

(pro-tip, you can use the `verticalArrangement=spaceBy` so you don't need to stick `Spacer`s between components everywhere)

```kotlin
Column(
            modifier = Modifier
                .fillMaxWidth()
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(8.dp),
        )
```

**Comment by jumaallan** [comment]
> ou don't need to stick `Spacer`s between components everywhere)

That will force us to have same spacing between compoenents, but we already do 16.dp anyway. Let me push a quick fix for that

**Review by vanshg — APPROVED** [review]
